src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ImmutableDispatchRuntime.cs

```diff
@@ -550,7 +553,7 @@ internal async Task<MessageRpc> ProcessMessageAsync(MessageRpc rpc)
                 rpc = await _authenticationBehavior.AuthenticateAsync(rpc);
             }
 
-            if (_authorizationBehavior != null)
+            if (_authorizationBehavior != null && !SupportsAuthorizationData)
             {
```

Comment:
If you remove the !SupportsAuthorizationData so that the original behavior still remains, I'd call this feature done and you can merge.

Reply:
@mconnew I removed the `!SupportsAuthorizationData` flag check but it had a side effect as `AuthorizationBehavior.AuthorizeAsync` internally set the ServiceSecurityContext which lead to the lost of the ClaimsPrincipal we previously stored during authentication. 
I then moved the call to `SetClaimsPrincipalToOperationContext` before the original call to `AuthorizationBehavior.AuthorizeAsync`.